File: app/channels/room_channel.rb
class RoomChannel < ApplicationCable::Channel
  def subscribed
    stream_from "room_channel_#{params['room']}"
  end


  def speak(data)
    if data['message'].include?('.jpg')
      kount = Message.count
      kount = kount.to_s
      puts "ここまではきてるよーーーーーーーーーーーー"
      File.binwrite("public/message_images/"+kount+".jpg", data["message"].read)
      if data["uuser"] != "nil"
        Message.create! content: kount+".jpg", user_id: params["user"], room_id: params['room']
      else
        Message.create! content: kount+".jpg", room_id: params['room']
      end
    else
      if data["uuser"] != "nil"
        Message.create! content: data['message'], user_id: params["user"], room_id: params['room']
      else
        Message.create! content: data['message'], room_id: params['room']
      end
    end
  end
end

File: app/javascript/channels/room_channel.js
import consumer from './consumer'

// $(function() { ... }); で囲むことでレンダリング後に実行される
// レンダリング前に実行されると $('#messages').data('room_id') が取得できない
// turbolinks を使っている場合は $(document).on('turbolinks:load', function() { ... }); で囲う
$(function() {
  const chatChannel = consumer.subscriptions.create({ channel: 'RoomChannel', room: $('#messages').data('room_id'),user: $('#messages').data('user_id') }, {
    connected() {
      // Called when the subscription is ready for use on the server
    },

    disconnected() {
      // Called when the subscription has been terminated by the server
    },

    received: function(data) {
      var user_data = $('#messages').data('user_id')
      if(user_data == "nil")
        return $('#messages').append(data);
      else
      return $('#messages').append(data)
    },

    speak: function(message) {
      console.log(message)
      return this.perform('speak', {
        message: message,
        uuser: $('#messages').data('user_id')
      });
    }
  });

  $(document).on('keypress', '[data-behavior~=room_speaker]', function(event) {
    if (event.keyCode === 13) {
      chatChannel.speak(event.target.value);
      event.target.value = '';
      $('#chats').animate({scrollTop: $('#chats')[0].scrollHeight}, 'normal');
      return event.preventDefault();
    }
  });
  $(window).on('load', function() {
    $('#chats').animate({scrollTop: $('#chats')[0].scrollHeight}, 'fast');
  });
});
